packages/remediations/src/steps/reviewSystems.js

```diff
@@ -52,22 +60,17 @@ const ReviewSystems = ({ issues, systems, registry, ...props }) => {
         }
     };
 
-    const onSelect = (selected) => {
-        input.onChange(selected);
-    };
-
     const onSelectRows = (value) => {
         dispatch({
             type: TOGGLE_BULK_SELECT,
             payload: value
         });
     };
 
-    const { selected, loaded, rows } = useSelector(({ entities }) => ({
-        selected: entities?.selected || [],
-        loaded: entities?.loaded,
-        rows: entities?.rows || []
-    }));
+    if (!isEqual(selectedRef.current, selected)) {
+        selectedRef.current = selected;
+        input.onChange(selected);
```

Comment:
This should be definitely in useEffect

Reply:
Not really, the `selected` is array we are checking is matching to previously selected systems. This actually has to be calculated on each render and if previously selected items are different from current one, the `input.onChange` has to tick.

Reply:
So just move the computation into the effect - useEffect is called on each render. You should not do sideeffects in render function.

Reply:
Good point, I'll use empty dependency for the effect and check in the body. Or we could use `useMemo` on the selected systems, but I'm not sure if that would help.

Reply:
Just for the record, using no dependency in the effect causes 1 render per select
```JS
useEffect(() => {
        if (!isEqual(input.value, selected)) {
            input.onChange(selected);
        }
    });
```
However using `isEqual(input.value, selected)` in the dependency list causes 2 renders per select action.
```JS
useEffect(() => {
  input.onChange(selected);
}, [ isEqual(input.value, selected) ]);
```